webknossos/tests/test_dataset.py

```diff
@@ -1562,7 +1562,56 @@ def test_add_copy_mag(data_format: DataFormat, output_path: Path) -> None:
     assert tuple(layer.bounding_box.topleft) == (6, 6, 6)
     assert tuple(layer.bounding_box.size) == (10, 20, 30)
 
-    copy_mag = layer.add_copy_mag(original_mag_2)
+    copy_mag = layer.add_copy_mag(original_mag_2, extend_layer_bounding_box=False)
+
+    assert (symlink_path / "color" / "1").exists()
+    assert len(layer._properties.mags) == 2
+
+    assert tuple(layer.bounding_box.topleft) == (6, 6, 6)
+    assert tuple(layer.bounding_box.size) == (10, 20, 30)
+
+    # Write data in copied layer
+    write_data = (np.random.rand(5, 5, 5) * 255).astype(np.uint8)
+    copy_mag.write(absolute_offset=(0, 0, 0), data=write_data)
+
+    assert np.array_equal(
+        copy_mag.read(absolute_offset=(0, 0, 0), size=(10, 10, 10))[0], write_data
+    )
+    assert np.array_equal(original_layer.get_mag(2).read()[0], original_data)
+
+    assure_exported_properties(ds)
+    assure_exported_properties(original_ds)
+
+
+@pytest.mark.parametrize("data_format,output_path", DATA_FORMATS_AND_OUTPUT_PATHS)
+def test_add_fs_copy_mag(data_format: DataFormat, output_path: Path) -> None:
+    ds_path = prepare_dataset_path(data_format, output_path, "original")
+    symlink_path = prepare_dataset_path(data_format, output_path, "with_symlink")
+
+    original_ds = Dataset(ds_path, voxel_size=(1, 1, 1))
+    original_layer = original_ds.add_layer(
+        "color", COLOR_CATEGORY, dtype_per_channel="uint8", data_format=data_format
+    )
+    original_layer.add_mag(1).write(
+        data=(np.random.rand(10, 20, 30) * 255).astype(np.uint8)
+    )
+    original_data = (np.random.rand(5, 10, 15) * 255).astype(np.uint8)
+    original_mag_2 = original_layer.add_mag(2)
```

Comment:
I think use can use this variable in line 1629 in place of `original_layer.get_mag(2)`.

Reply:
Tbh I just copied and adapted the previous test. At some point we'll have to refactor `test_dataset.py`, it has grown wild a bit^^ I tried to simplify the two touched test cases for now.